packages/icon/package.py

```diff
@@ -131,7 +135,7 @@ class Icon(AutotoolsPackage):
         description=
         'Enable extension of eccodes with center specific definition files')
 
-    depends_on('py-icon4py@0.0.1%gcc')
+    depends_on('py-icon4py@main%gcc')
```

Comment:
Better remove the `@version` completely and set it explicitely in an `spack.yaml`. This recipe needs to be as general as possible. Otherwise we can only build one specific version of icon-dsl